src/matlab2tikz.m

```diff
@@ -2378,6 +2395,9 @@ case size(cData,2) % specific y-data is given
             end
         case 3    % Text within a 3d plot
             pos = applyHgTransform(m2t, pos);
+            
+            %3d axes (need to set this for axes that contain only text
+            m2t.currentAxesContain3dData=true;
```

Comment:
Here is a test case that should reproduce the 3d-axis related bugs:

figure;
axes('view',[30,15],'DataAspectRatio',[0.5,1,1],'position',[0.1,0.1,0.8,0.8],'box','on’);
text(1,1,1,'test');
axes('position',[0.1,0.1538,0.8,0.6924],'box','on','color','none’);

(the line mentioned below does indeed not seem to be necessary anymore, I wrote this before that change was introduced)



For the transparency bug:

figure;
line([0.5,2.5],[0.5,2.5]);
hold on
imagesc([0,1;1,0],'AlphaData',uint8([100,200;100,200]));

The attached files show what the output should look like in matlab.

> On 2 Mar 2015, at 19:21, Oleg Komarov <notifications@github.com> wrote:
> 
> In src/matlab2tikz.m <https://github.com/matlab2tikz/matlab2tikz/pull/554#discussion_r25628613>:
> 
> > @@ -2378,6 +2395,9 @@ case size(cData,2) % specific y-data is given
> >              end
> >          case 3    % Text within a 3d plot
> >              pos = applyHgTransform(m2t, pos);
> > +            
> > +            %3d axes (need to set this for axes that contain only text
> > +            m2t.currentAxesContain3dData=true;
> We are not using that approach for 3D axis. We have now a direct check with the method isAxis3D(). Basically, these 3 lines are unnecessary.
> 
> —
> Reply to this email directly or view it on GitHub <https://github.com/matlab2tikz/matlab2tikz/pull/554/files#r25628613>.
>